src/unix/core.c

```diff
@@ -931,7 +934,7 @@ int uv__open_cloexec(const char* path, int flags) {
   int fd;
 
 #if defined(__linux__) || (defined(__FreeBSD__) && __FreeBSD__ >= 9) || \
-    defined(__DragonFly__)
+    defined(__DragonFly__) || (defined(__APPLE__) && defined(MAC_OS_X_VERSION_10_7))
```

Comment:
Maybe we could simplify that check by doing a `#if defined(UV__O_CLOEXEC)` instead.